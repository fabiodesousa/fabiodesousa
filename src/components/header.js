import React from 'react'
import PropTypes from 'prop-types'
import Headroom from 'react-headroom';
import { Link } from 'gatsby';
import './utility.css';
import { css } from "@emotion/core"
import Hamburger from './hamburger';

const header_links = [
  {to:'/', text:'Work'},
  {to:'/about', text:'About'},
  {to:'/moodboard', text:'Moodboard'}
]

const Header = ({ siteTitle, fontColor, color }) => (
  <Headroom disableInlineStyles>
    <div css={css`
      background-color: ${color};
    `}>
      <div
        css={css`
          margin-left: auto;
          margin-right: auto;
          display: flex;
          align-items: center;
          justify-content: center;
          height: 100px;
          width: 80vw;
          min-width: 350px;
          max-width: 1200px;
          position:relative;
          color: ${color === 'none' ? 'inherit' : fontColor};
        `}
      >
        <div css={css`
          margin-top: 2rem;
          display:none;
          @media(max-width:768px) {
            display: block;
          }
        `}>
          <Hamburger links={header_links} title={siteTitle} color={color} fontColor={fontColor} />
        </div>
        <div css={css`
          margin-right: auto;
          margin-top: 2rem;
          display: flex;
          width: 220px;
          justify-content: space-between;
          @media(max-width:768px) {
            display: none;
          }
          `
        }>
          {header_links.map((item, i) => (
            <span
              key={i}
              css={css`
                font-size: 0.8rem;
              `}
            >
              <Link key={item.to} to={item.to} css={css`text-decoration: none; color: inherit;`} className='hover-line-animate'>{item.text}</Link>
            </span>
          ))}
        </div>
        <div style={{
          display:'flex',
          margin: '1rem',
          position: 'fixed',
        }}>
          <h1 style={{
            margin:0,
            wrap:'no-wrap',
            fontSize: '3vmax'
          }}>
            <Link to="/" style={{textDecoration:'none', color:'inherit'}}>
              {siteTitle}
            </Link>
          </h1>
        </div>
      </div>
    </div>
  </Headroom>
)

Header.propTypes = {
  siteTitle: PropTypes.string,
  color: PropTypes.string,
  fontColor: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
