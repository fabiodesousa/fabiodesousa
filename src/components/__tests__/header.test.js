import React from "react"
import renderer from "react-test-renderer"
import {PureHeader as Header} from "../header"



describe("Header", () => {
  it("renders without crashing", () => {
    const tree = renderer
    .create(
      <Header siteTitle="Tampa Bay" backgroundColor='white'>
        <div>foo</div>
        <div>bar</div>
      </Header>
    )
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
})