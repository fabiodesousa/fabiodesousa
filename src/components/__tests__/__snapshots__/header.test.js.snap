// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "white",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": "4rem",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0",
        "padding": "1.45rem 1.0875rem",
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "color": "#3b383f",
          "textDecoration": "none",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Tampa Bay
      </h1>
    </a>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "margin": "0",
        "padding": "1rem",
      }
    }
  >
    <div>
      foo
    </div>
    <div>
      bar
    </div>
  </div>
</div>
`;
